src/sed/loader/flash/buffer_handler.py

```diff
@@ -124,13 +110,35 @@ def __init__(
         self.n_cores: int = config["core"].get("num_cores", os.cpu_count() - 1)
         self.fp: BufferFilePaths = None
         self.df: dict[str, dd.DataFrame] = {typ: None for typ in DF_TYP}
+        fill_formats = self._config.get("fill_formats")
         self.fill_channels: list[str] = get_channels(
             self._config,
-            ["per_pulse", "per_train"],
+            fill_formats,
             extend_aux=True,
         )
         self.metadata: dict = {}
 
+        core_beamline = config["core"].get("beamline")
+        self.DataFrameCreator: type[BaseDataFrameCreator] = None
+        if core_beamline == "pg2":
+            self.DataFrameCreator = DataFrameCreator
+        elif core_beamline == "cfel":
+            self.DataFrameCreator = CFELDataFrameCreator
+        else:
+            raise ValueError(f"Unsupported core beamline: {core_beamline}")
+
+    def _validate_h5_files(self, config, h5_paths: list[Path]) -> list[Path]:
```

Comment:
This validation was previously in BufferFilePaths and we had a discussion to move it from there. I find this location better (also was necessary due to restructure)